src/components/vega-lite/index.tsx

```diff
@@ -53,13 +57,17 @@ export class VegaLite extends React.PureComponent<VegaLiteProps, any> {
     try {
       const spec = vl.compile(vlSpec, logger).spec;
       const runtime = vega.parse(spec, vlSpec.config);
-      const view = new vega.View(runtime)
+      this.view = new vega.View(runtime)
         .logLevel(vega.Warn)
         .initialize(this.refs[CHART_REF] as any)
         .renderer(this.props.renderer || 'svg')
-        .hover()
-        .run();
-      vegaTooltip.vega(view);
+        .hover();
+      vegaTooltip.vega(this.view);
+      if (isInlineData(this.props.data) && isNamedData(vlSpec.data)) {
```

Comment:
We should separate the binding method from the compile method.  
(Suppose `spec` is the same and `data` changes?)